### DISCUSSION TITLE
Use Hono routing directly?

### DISCUSSION THREAD

**Discussion by alexturpin** [marbemac/ssrx#42] [Q&A]
Hello everyone,

Just found this package, been looking for a long time for a lightweight solution to render React applications on Cloudflare workers. I'm trying to see if I can integrate ssrx by using Hono's routing directly. For instance:

```
app.get("/", async (c) => {
  return renderPage(c.req, <Home />) // Or maybe a middleware or something on context
})
```

However I'm not too sure what the best practices around that would be; right now I export `createApp`:
```
export const { clientHandler, serverHandler, ctx } = createApp({
  appRenderer:
    ({ req }) =>
    () =>
      (
        <div>
          My App <Counter />
        </div>
      ),
  plugins: [assetsPlugin()],
})
```

And can always render that simple app in my route:

```
app.get("/react", async (c) => {
  const { stream, statusCode } = await serverHandler({
    req: c.req.raw,
  })

  return new Response(stream, { status: statusCode(), headers: { "Content-Type": "text/html" } })
})
```

But I'm wondering if people have any thoughts on best practices / a neat way to wrap this up so I can use it directly with Hono's routing. Should I write a plugin? Will I be unable to benefit from code splitting if I do it this way, and I should just swallow my pride and use react-router for those routes?

Thanks for any feedback you might have!

**Comment by marbemac** [comment]
Hmm, I guess there is kind of an expectation that one is using a client side router. For your use case, how are you planning to do routing on the client side? A full page reload on every link click?

**Reply by alexturpin** [reply]
Yeah, regular links, I have an SPA within my app with its own client-side router but I'm mostly interested in ssrx to render the "top level pages" of my app, and that can be MPA, I just need React SSR and hydration. If using a router is the recommended way though I will do that 🤷‍♂️

**Reply by marbemac** [reply]
I see - is it a homegrown client-side router? And what kind of "top level pages"? Trying to wrap my head around the use case :).

**Reply by alexturpin** [reply]
Absolutely, I appreciate the time, I'd love to elaborate.

My overall app uses React for a marketing website with several "top level" pages (home, about, terms and conditions), as well as a specific high interactivity "app" page, call it `/app`. That app page has its own client-side router (`wouter`) with a base of `/app`, so it manages the SPA-ness routing I need there. 

Right now I'm using Astro which cannot render my React UI framework correctly using SSR, so I'm exploring other solutions. With ssrx so far it sounds like I would use something like react-router to do that "top level pages" routing, and I can still have my little client-side routing separate for my app.

**Reply by marbemac** [reply]
Ah yeah I see. Yes w SSRx the best option is probably to just use something like react-router to do the top level routing. That said, pages like about, terms of service, etc, are particularly well suited to server components - have you considered trying to embed your wouter app into a nextjs app? Where next handles the shell and marketing pages, and renders your wouter app at /app/*?

I'm not a huge nextjs fan myself, BUT it might be a decent alternative option for your use case.

**Reply by alexturpin** [reply]
Thanks for the suggestion! I'm really not a fan of Next in general, and especially on Cloudflare Workers I find it doesn't make a lot of sense. Compatibility is better but Next.js is still very heavy from a server code perspective in an environment where size matters. Additionally I'd rather be more in control seeing as emulating all Cloudflare functionality locally hasn't always been straightforward. Thank you for all the feedback!

**Reply by alexturpin** [reply]
The ergonomics of react-router v7 are actually looking even nicer, so combining that with @ssrx might just be the overall killer solution I'd been looking for!
